### PR TITLE
Fix rspec tests broken by ruby update and changing the directory.

### PR DESCRIPTION
**Pull request by tirodkar**
Movin' the tmp directory from something we create to a fixed one.

Required for https://github.com/google/EarlGrey/pull/305
Thanks @bootstraponline

### CODE REVIEW

File: gem/spec/spec_helper.rb [hunk 1]
```diff
@@ -65,29 +65,28 @@ def diff_project(project_init, project_after, command_array)
     raise 'command_array is not an array' unless command_array && command_array.is_a?(Array)
     # dirname for "/fixtures/project_before/." => /fixtures/project_before"
     xcodeproj_2 = File.join(File.dirname(project_after), 'Example.xcodeproj')
+    tmp_dir = File.join(project_init, '../tmpdir')
+    puts tmp_dir
```
Comment (bootstraponline) [inline gem/spec/spec_helper.rb]: probably want to remove this debugging line?

File: gem/spec/spec_helper.rb [hunk 2]
```diff
@@ -65,29 +65,28 @@ def diff_project(project_init, project_after, command_array)
...
-          puts diff
-          raise 'difference detected'
-        end
+      contains_swift_2 = command_array.include? '--swift_version=2.2'
+      diff = ProjectDiff.run(xcodeproj_1, xcodeproj_2, contains_swift_2)
+      if diff != NIL_YAML
+        puts diff
+        raise 'difference detected'
       end
+      FileUtils.rm_rf(Dir.glob(File.join(tmp_dir, "*")).reject { |file| file.end_with?("README")})
```
Comment (bootstraponline) [inline gem/spec/spec_helper.rb]: cleanup should always happen in an `ensure` block

Comment (tirodkar) [inline gem/spec/spec_helper.rb]: Done.

File: gem/spec/spec_helper.rb (hunk 1 above)
Comment (tirodkar) [inline gem/spec/spec_helper.rb]: Done.

**Comment by bootstraponline** [comment]
looks good! 👍

**Pull request merged**